### ISSUE TITLE
Post distributed to deleted site causes error 500 on edit

### ISSUE THREAD

**Issue by gaelgerard** [10up/distributor#208]
Hello,

Thank you for your plugin wich is very usefull for me.

However I have an issue on deleted sites of my network.

Posts distribution is not updated when I delete websites.

I don't know if it is the same on default wp posts because I only test on custom post type.

I have 4 websites in my network.

I distributed a couple of custom posts to those sites.

When I delete one of those websites, those shared posts keep displaying "Distributed to 4 connections (?)" while there are only 3 websites left.

Moreover, when I try to update those distributed posts, wp gives me a beautifull white 500 error page due to fatal error finding the deleted site.

Could that be a problem with my custom post type?

Thank you for your help.

Gaël

**Comment by dkotter** [comment]
@gaelgerard Thanks for the report! I was able to reproduce this fairly easily and was able to get to the root of the issue. Basically when a post (or other content type) is updated, we try and push those updates to any sites that have been distributed to.

We weren't checking if the site actually exists anymore, so we end up trying to push to a non-existent site and that would cause an error. I just pushed a fix for this, that checks if a site still exists before trying to push an update. Feel free to test that PR out if you want.
